src/PointwiseFunctions/Hydro/EquationsOfState/Enthalpy.cpp

```diff
@@ -0,0 +1,451 @@
+template <class DataType>
+Scalar<DataType>
+Enthalpy<LowDensityEoS>::kappa_times_p_over_rho_squared_from_density_impl(
+    const Scalar<DataType>& rest_mass_density) const {
+  return make_with_value<Scalar<DataType>>(get(rest_mass_density), 0.0);
+}
+template <typename LowDensityEoS>
+double Enthalpy<LowDensityEoS>::chi_from_density(
+    const double rest_mass_density) const {
+  if (Enthalpy::in_low_density_domain(rest_mass_density)) {
+    return get(
+        low_density_eos_.chi_from_density(Scalar<double>(rest_mass_density)));
+  } else {
+    const double x = x_from_density(rest_mass_density);
+    return evaluate_coefficients(derivative_coefficients_, x);
+  }
+}
+
+template <typename LowDensityEoS>
+double Enthalpy<LowDensityEoS>::specific_internal_energy_from_density(
+    const double rest_mass_density) const {
+  if (Enthalpy::in_low_density_domain(rest_mass_density)) {
+    return get(low_density_eos_.specific_internal_energy_from_density(
+        Scalar<double>(rest_mass_density)));
+  } else {
+    return 1 / rest_mass_density *
+               energy_density_from_log_density(
+                   x_from_density(rest_mass_density)) -
+           1.0;
+  }
+}
+template <typename LowDensityEoS>
+double Enthalpy<LowDensityEoS>::specific_enthalpy_from_density(
+    const double rest_mass_density) const {
+  if (Enthalpy::in_low_density_domain(rest_mass_density)) {
+    return get(low_density_eos_.specific_enthalpy_from_density(
+        Scalar<double>(rest_mass_density)));
+  } else {
+    return evaluate_coefficients(coefficients_,
+                                 x_from_density(rest_mass_density));
+  }
+}
+
+// P(x) = rho(x)h(x) - e(x) with h the specific enthalpy, and e
+// the energy density.
+template <typename LowDensityEoS>
+double Enthalpy<LowDensityEoS>::pressure_from_log_density(
+    const double x) const {
+  auto rest_mass_density = density_from_x(x);
+  if (in_low_density_domain(rest_mass_density)) {
+    return get(low_density_eos_.pressure_from_density(
+        Scalar<double>(rest_mass_density)));
+  } else {
+    return rest_mass_density * evaluate_coefficients(coefficients_, x) -
+           energy_density_from_log_density(x);
+  }
+}
+template <typename LowDensityEoS>
+double Enthalpy<LowDensityEoS>::pressure_from_density(
+    const double rest_mass_density) const {
+  if (in_low_density_domain(rest_mass_density)) {
+    return get(low_density_eos_.pressure_from_density(
+        Scalar<double>(rest_mass_density)));
+  } else {
+    const double x = x_from_density(rest_mass_density);
+    return pressure_from_log_density(x);
+  }
+}
+
+// Solve for h(rho)=h0, which requires rootfinding for this EoS
+template <typename LowDensityEoS>
+double Enthalpy<LowDensityEoS>::rest_mass_density_from_enthalpy(
+    const double specific_enthalpy) const {
+  if (specific_enthalpy <= minimum_enthalpy_) {
+    return get(low_density_eos_.rest_mass_density_from_enthalpy(
+        Scalar<double>(specific_enthalpy)));
```

Comment:
There are a few of these warnings that things aren't covered in the test. If the warnings are correct, could you add some explicit tests for these calls below the min thresholds?